client/src/components/Sidebar/Sidebar.js

```diff
@@ -21,9 +23,25 @@ const useStyles = makeStyles(() => ({
 
 const Sidebar = (props) => {
   const classes = useStyles();
+  const dispatch = useDispatch();
+
+  const [newMessagesConvoId, setNewMessagesConvoId] = useState('')
+  
   const conversations = props.conversations || [];
+  const newMessageConvoId = props.newMessageConvoId
   const { handleChange, searchTerm } = props;
 
+  useEffect(() => {
+    setNewMessagesConvoId(newMessageConvoId)
+  }, [newMessageConvoId])
+
+  useEffect(() => {
+    const convoIndex = conversations.findIndex(convo => convo.id === newMessageConvoId)
+    if(convoIndex > 0) {
+      dispatch(moveConvoToTop(newMessagesConvoId))
+    }
+  }, [newMessagesConvoId])
```

Comment:
Hmm... this is quite inefficient since there's a loop inside the component which will run every time that a new message is received.
- This functionality should be handled through the reordering of conversations + socket emission when a new message is received to alleviate component logic